client/ayon_deadline/plugins/publish/global/collect_jobinfo.py

```diff
@@ -0,0 +1,301 @@
+from ayon_core.lib.profiles_filtering import filter_profiles
+from ayon_core.addon import AddonsManager
+
+from ayon_deadline.lib import (
+    FARM_FAMILIES,
+    AYONDeadlineJobInfo,
+)
+
+
+class CollectJobInfo(pyblish.api.InstancePlugin, AYONPyblishPluginMixin):
+    """Collect variables that belong to Deadline's JobInfo.
+
+    Variables like:
+    - department
+    - priority
+    - chunk size
+
+    """
+
+    order = pyblish.api.CollectorOrder + 0.420
+    label = "Collect Deadline JobInfo"
+
+    families = FARM_FAMILIES
+    targets = ["local"]
+
+    profiles = []
+    pool_enum_values = []
+
+    def process(self, instance):
+        attr_values = self._get_jobinfo_defaults(instance)
+
+        attr_values.update(self.get_attr_values_from_data(instance.data))
+        job_info = AYONDeadlineJobInfo.from_dict(attr_values)
+
+        self._handle_machine_list(attr_values, job_info)
+
+        self._handle_additional_jobinfo(attr_values, job_info)
+
+        instance.data["deadline"]["job_info"] = job_info
+
+        # pass through explicitly key and values for PluginInfo
+        plugin_info_data = None
+        if attr_values["additional_plugin_info"]:
+            plugin_info_data = (
+                json.loads(attr_values["additional_plugin_info"]))
+        instance.data["deadline"]["plugin_info_data"] = plugin_info_data
+
+    def _handle_additional_jobinfo(self,attr_values, job_info):
+        """Adds not explicitly implemented fields by values from Settings."""
+        additional_job_info = attr_values["additional_job_info"]
+        if not additional_job_info:
+            return
+        for key, value in json.loads(additional_job_info).items():
+            setattr(job_info, key, value)
+
+    def _handle_machine_list(self, attr_values, job_info):
+        machine_list = attr_values["machine_list"]
+        if machine_list:
+            if job_info.MachineListDeny:
+                job_info.Blacklist = machine_list
+            else:
+                job_info.Whitelist = machine_list
+
+    @classmethod
+    def apply_settings(cls, project_settings):
+        settings = project_settings["deadline"]
+        profiles = settings["publish"][cls.__name__]["profiles"]
+
+        cls.profiles = profiles or []
+
+        addons_manager = AddonsManager()
+        deadline_addon = addons_manager["deadline"]
+        deadline_server_name = settings["deadline_server"]
+        dl_server_info = deadline_addon.deadline_servers_info.get(
+                deadline_server_name)
+
+        auth = (dl_server_info["default_username"],
+                dl_server_info["default_password"])
+        pools = deadline_addon.get_deadline_pools(
+            dl_server_info["value"],
+            auth
+        )
+        for pool in pools:
+            cls.pool_enum_values.append({"value": pool, "label": pool})
+
+    @classmethod
+    def get_attr_defs_for_instance(cls, create_context, instance):
+        host_name = create_context.host_name
```

Comment:
I'd say that even BEFORE we get to profile filtering below we should first already filter to `families` itself. It makes no sense to expose attributes for families for which this plug-in does not run to begin with?

Reply:
I was testing this PR and came across this. 
tbh, it's confusing to have attributes for instances that are not assigned to the plugin.

Reply:
If it doesn't run, this wont be triggered at all or will be?

Reply:
It will be. It runs for ALL instances - even those that do not match by `families`.

Reply:
```suggestion
        if not cls.instance_matches_plugin_families(instance):
            return []

        host_name = create_context.host_name
```

Reply:
> It will be. It runs for ALL instances - even those that do not match by families. It's up to the function to define the filtering.
> 
> The reasoning behind that I suppose is up to @iLLiCiTiT

So publish plugins can show attributes based on different logic than families.